python/src/iceberg/catalog/base.py

```diff
@@ -0,0 +1,223 @@
+    ) -> Table:
+        """Create a table
+
+        Args:
+            namespace: A tuple of table's namespace levels. Ex: ('com','org','dept')
+            name: Table name
+            schema: Table's schema.
+            location: Location for the table. Optional Argument.
+            partition_spec: PartitionSpec for the table. Optional Argument.
+            properties: Table metadata that can be a string based dictionary. Optional Argument.
+
+        Returns:
+            Table: the created table instance
+
+        Raises:
+            AlreadyExistsError: If a table with the name already exists
+        """
+
+    @abstractmethod
+    def table(self, namespace: Tuple[str, ...], name: str) -> Table:
+        """Loads the table's metadata and returns the table instance.
+
+        You can also use this method to check for table existence using 'try catalog.table() except TableNotFoundError'
+        Note: This method does not load table's data in any form.
+
+        Args:
+            namespace: A tuple of table's namespace levels. Ex: ('com','org','dept')
+            name: Table's name.
+
+        Returns:
+            Table: the table instance with its metadata
+
+        Raises:
+            TableNotFoundError: If a table with the name does not exist
+        """
+
+    @abstractmethod
+    def drop_table(self, namespace: Tuple[str, ...], name: str, purge: bool = True) -> None:
+        """Drop a table; Optionally purge all data and metadata files.
+
+        Args:
+            namespace: A tuple of table's namespace levels. Ex: ('com','org','dept')
+            name: table name
+            purge: Defaults to true, which deletes all data and metadata files in the table; Optional Argument
+
+        Raises:
+            TableNotFoundError: If a table with the name does not exist
+        """
+
+    @abstractmethod
+    def rename_table(self, from_namespace: Tuple[str, ...], from_name: str, to_namespace: Tuple[str, ...], to_name: str) -> Table:
+        """Rename a fully classified table name
+
+        Args:
+            from_namespace: Existing table's namespace. A tuple of table's namespace levels. Ex: ('com','org','dept')
+            from_name: Existing table's name.
+            to_namespace: New Table namespace to be assigned. Tuple of namespace levels. Ex: ('com','org','new')
+            to_name: New Table name to be assigned.
+
+        Returns:
+            Table: the updated table instance with its metadata
+
+        Raises:
+            TableNotFoundError: If a table with the name does not exist
+        """
+
+    @abstractmethod
+    def replace_table(
+        self,
+        *,
+        namespace: Tuple[str, ...],
+        name: str,
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: Optional[PartitionSpec] = None,
+        properties: Optional[Dict[str, str]] = None,
+    ) -> Table:
+        """Starts a transaction and replaces the table with the provided spec.
```

Comment:
It doesn't look like this starts a transaction because the return type is a `Table`. For now, how about removing this method until we want to add transactions? We can get a lot done with just `create_table` and `drop_table`. Transactions aren't really ready to be added yet anyway.

Reply:
Ack. Will remove this API for now.